clients/src/main/java/org/apache/kafka/clients/admin/DeleteConsumerGroupOffsetsResult.java

```diff
@@ -46,39 +47,43 @@
     public KafkaFuture<Void> partitionResult(final TopicPartition partition) {
         final KafkaFutureImpl<Void> result = new KafkaFutureImpl<>();
 
-        this.future.whenComplete(new BiConsumer<Map<TopicPartition, Errors>, Throwable>() {
-            @Override
-            public void accept(final Map<TopicPartition, Errors> topicPartitions, final Throwable throwable) {
-                if (throwable != null) {
-                    result.completeExceptionally(throwable);
-                } else if (!topicPartitions.containsKey(partition)) {
-                    result.completeExceptionally(new IllegalArgumentException(
-                        "Group offset deletion for partition \"" + partition +
-                        "\" was not attempted"));
+        this.future.whenComplete((topicPartitions, throwable) -> {
+            if (throwable != null) {
+                result.completeExceptionally(throwable);
+            } else {
+                Throwable exception = KafkaAdminClient.getSubLevelError(topicPartitions, partition,
+                    "Group offset deletion for partition \"" + partition + "\" was not attempted");
+                if (exception != null) {
+                    result.completeExceptionally(exception);
                 } else {
-                    final Errors error = topicPartitions.get(partition);
-                    if (error == Errors.NONE) {
-                        result.complete(null);
-                    } else {
-                        result.completeExceptionally(error.exception());
-                    }
+                    result.complete(null);
                 }
-
             }
         });
-
         return result;
     }
 
     /**
      * Return a future which succeeds only if all the deletions succeed.
      */
     public KafkaFuture<Void> all() {
-        return this.future.thenApply(new BaseFunction<Map<TopicPartition, Errors>, Void>() {
-            @Override
-            public Void apply(final Map<TopicPartition, Errors> topicPartitionErrorsMap) {
-                return null;
+        final KafkaFutureImpl<Void> result = new KafkaFutureImpl<>();
+
+        this.future.whenComplete((topicPartitions, throwable) -> {
+            if (throwable != null) {
+                result.completeExceptionally(throwable);
+            } else {
+                for (TopicPartition partition : partitions) {
+                    Throwable exception = KafkaAdminClient.getSubLevelError(topicPartitions, partition,
+                        "Group offset deletion for partition \"" + partition + "\" was not attempted");
+                    if (exception != null) {
+                        result.completeExceptionally(exception);
```

Comment:
FYI - This will likely impact the behavior of the consumer group command which expect only group level errors to be raised: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala#L440

Reply:
That's right, what's your opinion on whether we should export the member level error? As I could imagine other implementation using Java admin client may expect something different (like `all` could tell us if there are any sub level failures)